src/util/string.cpp

```diff
@@ -3,3 +3,14 @@
 // file COPYING or http://www.opensource.org/licenses/mit-license.php.
 
 #include <util/string.h>
+#include <util/spanparsing.h>
+
+std::vector<std::string> SplitString(const std::string& str, char sep)
```

Comment:
Seems like a nice way to solve this! Unfortunately, I fear my C++ template skills are currently too lousy to come up with this. Will give it a try on the weekend. That said, I'm also happy to close this PR for another one with a better solution :)

Reply:
I think moving the code into the header and changing it into something like that should work:

```cpp
template <typename T = Span<const char>>
std::vector<T> Split(const Span<const char>& sp, char sep)
{
    std::vector<T> ret;
    auto it = sp.begin();
    auto start = it;
    while (it != sp.end()) {
        if (*it == sep) {
            ret.emplace_back(start, it);
            start = it + 1;
        }
        ++it;
    }
    ret.emplace_back(start, it);
    return ret;
}
```

So the code is practically identical, only the return take is now a template argument. Note that I'm typing this on my phone because I don't have access to a computer the next few days so this probably won't compile :)

Reply:
If you decide to do that, it would be possible to define an alias like so:

```cpp
constexpr auto SplitString = &spanparsing::Split<std::string>;

Reply:
@martinus: Thanks, I will try that out in a bit!
@MarcoFalke: That's a good idea with the alias, then I only need to change the implementation and keep the rest as it is now.